pkg/models/sigv4.go

```diff
@@ -97,10 +97,8 @@ func (m *SigV4Middleware) signer() (*v4.Signer, error) {
 		c = credentials.NewStaticCredentials(m.Config.AccessKey, m.Config.SecretKey, "")
 	case Credentials:
 		c = credentials.NewSharedCredentials("", m.Config.Profile)
-	}
-
-	// passing nil credentials will force AWS to allow a more complete credential chain vs the explicit default
-	if c == nil {
+	case Default:
+		// passing nil credentials will force AWS to allow a more complete credential chain vs the explicit default
```

Comment:
👍 
I guess you have to add also a default case for any other authentication type, otherwise the exhaustive linter would complain?

Reply:
@papagian Actually it doesn't, since it can tell there are no more cases.

Reply:
The default of the linter is actually to require that all switch branches are covered, you have to configure it to accept `default` as fallback.

Reply:
GTK, thanks!